From: =?utf-8?b?RnLDqWTDqXJpYyBCb25uYXJk?= <frediz@debian.org>
Date: Mon, 22 Feb 2021 09:14:58 +0100
Subject: Additions to apparmor profiles
MIME-Version: 1.0
Content-Type: text/plain; charset="utf-8"
Content-Transfer-Encoding: 8bit

o Debian uses dash as its default shell
o Debian has it's git components under /usr/lig/git-core/
o /usr/bin/git is being called directly when scm=git
o unzip is called just 'unzip' on Debian
o we want to be able to run dpkg -l in the worker to log installed
  packages.
o git takes a look at /etc/mailname for creating commit messages
  (perhaps it wouldn't if the Author were explicitly set, so maybe this
  is not really needed?)
o git-remote-{http*,rm} get used for pushing changes to needles,
  if enabled
o kvm uses seabios to boot
o needles need to be writable to enable editing

Author: Philip Hands <phil@hands.com>,
        Frédéric Bonnard <frediz@debian.org>
---
 profiles/apparmor.d/usr.share.openqa.script.openqa | 24 ++++++++++++++--------
 profiles/apparmor.d/usr.share.openqa.script.worker |  9 ++++++--
 2 files changed, 23 insertions(+), 10 deletions(-)

diff --git a/profiles/apparmor.d/usr.share.openqa.script.openqa b/profiles/apparmor.d/usr.share.openqa.script.openqa
index 7966abf..6642731 100644
--- a/profiles/apparmor.d/usr.share.openqa.script.openqa
+++ b/profiles/apparmor.d/usr.share.openqa.script.openqa
@@ -20,19 +20,25 @@
   /etc/openqa/openqa.ini r,
   /etc/openqa/templates/ r,
   /etc/openqa/templates/** r,
+  /etc/mailname r,
   /proc/meminfo r,
   /tmp/** rwk,
+  /usr/bin/dash ix,
   /usr/bin/perl ix,
   /usr/bin/ssh rcx,
   /usr/bin/timeout rix,
-  /usr/bin/unzip-plain rix,
-  /usr/lib/git/git rix,
-  /usr/lib/git/git-add rix,
-  /usr/lib/git/git-commit rix,
-  /usr/lib/git/git-push rix,
-  /usr/lib/git/git-rebase rix,
-  /usr/lib/git/git-remote rix,
-  /usr/lib/git/git-rm rix,
+  /usr/bin/unzip{,-plain} rix,
+  /usr/bin/git rix,
+  /usr/lib/git{,-core}/git rix,
+  /usr/lib/git{,-core}/git-add rix,
+  /usr/lib/git{,-core}/git-commit rix,
+  /usr/lib/git{,-core}/git-push rix,
+  /usr/lib/git{,-core}/git-rebase rix,
+  /usr/lib/git{,-core}/git-remote rix,
+  /usr/lib/git{,-core}/git-remote-http rix,
+  /usr/lib/git{,-core}/git-remote-https rix,
+  /usr/lib/git{,-core}/git-rm rix,
+  /usr/lib/git{,-core}/git-upload-pack rix,
   /usr/share/openqa/assets/** r,
   /usr/share/openqa/dbicdh/** r,
   /usr/share/openqa/lib/** r,
@@ -90,6 +96,8 @@
 
   }
 
+  /etc/ssl/openssl.cnf r,
+
   # Site-specific additions and overrides. See local/README for details.
   #include <local/usr.share.openqa.script.openqa>
 }
diff --git a/profiles/apparmor.d/usr.share.openqa.script.worker b/profiles/apparmor.d/usr.share.openqa.script.worker
index 9822dd4..bfd0d06 100644
--- a/profiles/apparmor.d/usr.share.openqa.script.worker
+++ b/profiles/apparmor.d/usr.share.openqa.script.worker
@@ -26,7 +26,7 @@
 
   network netbeui raw,
 
-  /{usr/,}bin/bash rix,
+  /{usr/,}bin/{b,d}ash rix,
   /{usr/,}bin/pkill rix,
   /dev/ r,
   /dev/bus/usb/ r,
@@ -36,6 +36,7 @@
   /dev/ptmx rw,
   /dev/pts/* rw,
   /dev/tty rw,
+  /etc/dpkg/** r,
   /etc/libnl/classid r,
   /etc/openqa/client.conf r,
   /etc/openqa/workers.ini r,
@@ -64,6 +65,7 @@
   /sys/devices/system/node/** r,
   /sys/firmware/devicetree/base/ r,
   /sys/fs/cgroup/systemd/openqa.slice/** rw,
+  /sys/kernel/mm/transparent_hugepage/enabled r,
   /tmp/* rwk,
   /usr/bin/Xvnc rCx,
   /usr/bin/cat rix,
@@ -72,6 +74,7 @@
   /usr/bin/cp rix,
   /usr/bin/date rix,
   /usr/bin/dirname rix,
+  /usr/bin/dpkg{,-query} rix,
   /usr/bin/eatmydata rix,
   /usr/bin/env rix,
   /usr/bin/find rix,
@@ -125,7 +128,9 @@
   /usr/share/git-core/templates/ r,
   /usr/share/git-core/templates/** r,
   /usr/share/qemu/* rk,
+  /usr/share/seabios/* r,
   /usr/share/qemu/keymaps/* r,
+  /var/lib/dpkg/** r,
   /var/lib/openqa/cache/ r,
   /var/lib/openqa/cache/** rwk,
   /var/lib/openqa/pool/ r,
@@ -141,9 +146,9 @@
   /var/lib/openqa/share/factory/tmp/ rw,
   /var/lib/openqa/share/factory/tmp/** rw,
   /var/lib/openqa/share/tests/** r,
+  /var/lib/openqa/share/tests/*/needles/** rw,
   owner /sys/**/ rw,
 
-
   profile /usr/bin/Xvnc {
     #include <abstractions/X>
     #include <abstractions/base>
